File: java/1304_find_n_unique_integers_sum_up_to_zero.java
"""
Problem: 1304 - Find N Unique Integers Sum up to Zero  
Description: Given an integer n, return any array containing n unique integers such that their sum is 0.  
Date: 2025-09-07
"""

class Solution {
    public int[] sumZero(int n) {
        int [] nums = new int [n];
        
        if (n % 2 == 0){
            fill(nums,true,n);
        }
        else {
            fill(nums,false,n);
        }
        return nums;
    }

    private void fill(int[] nums, boolean flag,int n){
        int j = 0;
        for (int i = -n/2 ; i <= n/2 ;i++){
            if(flag && i == 0) continue ;
            nums[j] = i;
            j++;
        }
        return;
    }
}

File: python/1304_find_n_unique_integers_sum_up_to_zero.py
"""
Problem: 1304 - Find N Unique Integers Sum up to Zero  
Description: Given an integer n, return any array containing n unique integers such that their sum is 0.  
Date: 2025-09-07
"""

class Solution:
    def sumZero(self, n: int) -> List[int]:
        nums = []
        self.fill(nums, n)
        if n % 2 :
            nums.append(0)
        return nums

    def fill(self, nums: list, n: int) :
        for i in range(1,n//2 + 1):
            nums.append(-i)
            nums.append(i)

        return
